docs/authoring.md

```diff
@@ -76,13 +76,14 @@ In case you have more than one template and want to reuse filters, you can put t
 
 ## Hooks
 
-Hooks are functions called by the generator on a specific moment in the generation process.
+Hooks are functions called by the generator on a specific moment in the generation process. These hooks can have arguments provided to them or being expected to return a value.
 The following types of hooks are currently supported:
 
-|Hook name|Description|
-|---|---|
-| `generate:before` | Called after registration of all filters and before generator starts processing of the template.|
-| `generate:after` | Called at the very end of the generation. |
+|Hook name|Description| Return type | Arguments 
+|---|---|---|---|
+| `generate:before` | Called after registration of all filters and before generator starts processing of the template.| void | none
+| `generate:after` | Called at the very end of the generation. | void | none
+| `generate:changeFilename` | Called right before saving a file template. | string : a new filename for the generator to use for the file template | { "filename" : string }
```

Comment:
Maybe a small improve for description?
`Called right before saving a new file generated by [file template](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#file-templates)`
And perhaps `{ "originalFilename" : string }` instead of `{ "filename" : string }` ?

Reply:
I like it 👍